sqlglot/expressions.py

```diff
@@ -6749,7 +6748,7 @@ class UnixSeconds(Func):
 
 
 class Uuid(Func):
-    _sql_names = ["UUID", "GEN_RANDOM_UUID", "GENERATE_UUID", "UUID_STRING"]
+    _sql_names = ["UUID", "GEN_RANDOM_UUID", "GENERATE_UUID", "NEWID", "UUID_STRING"]
```

Comment:
Having to touch this seemed a little bit weird. Not sure if I missed something in the specific dialect implementations.

Reply:
You can either do this or add `"NEWID": exp.Uuid.from_arg_list` in `TSQL.Parser.FUNCTIONS`. The latter is safer, because the approach in this PR may have silent side-effects if there are other dialects that support `NEWID(...)` and, e.g., it expects a different number of arguments than what the expression supports.

My hunch is this should be safe here, but could you quickly check if this function exists in other dialects and whether its semantics / argument count matches what we have here?

Reply:
Okay, that's much nicer. I went ahead and made that change in https://github.com/tobymao/sqlglot/pull/4718/commits/293edfee92bec60ec112d272cac40308ac105e77